tests/wpcom-vip-two-factor/sms-provider.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+class Two_Factor_SMS__Test extends \WP_UnitTestCase {
+ public function setUp() {
+		require_once __DIR__ . '/../wpcom-vip-two-factor/sms-provider.php';
```

Comment:
This path needs one additional directory traversal.
```suggestion
		require_once __DIR__ . '/../../wpcom-vip-two-factor/sms-provider.php';
```

Reply:
Argh, didn't realize it's loaded as a plugin :bangs-head:. Ahh okay, you need to require both paths b/c `sms-provider.php` relies on `two-factor.php` in the shared plugins.

---
Just noting the original issue I had with testing for reference:

Following: https://phpunit.readthedocs.io/en/9.0/test-doubles.html#test-doubles

`Error: Class 'Automattic\VIP\TwoFactor\Two_Factor_Provider' not found`

or some variation of that. I tried adding a namespace (`Automattic\VIP\TwoFactor`) in the original class, then calling that, calling the regular class name, etc., but it wont recognize it.

--

I changed the code to use `createMock()` instead of `getMockBuilder` but I get:
`Error: Call to undefined method Mock_Two_Factor_SMS_383faa33::process_results()`

or

`Cannot stub or mock class or interface "Two_Factor_SMS" which does not exist
I don’t know why it’s not recognizing the class`

Reply:
Also, regarding the directory traversal, I initially had it set to
`/../../wpcom-vip-two-factor/sms-provider.php` (the suggested change),
but I noticed in this code, the traversal is missing one step:

https://github.com/Automattic/vip-go-mu-plugins/blob/master/tests/elasticsearch/includes/classes/test-class-healthjob.php#L7

If you try to run it with the correct path, it errors out. Not sure what that's about

Reply:
As we went through  today, this is because we were never actually calling any methods, just setting up the mock.